### PR TITLE
Upgrade webpack to v4.11.1

### PR DESCRIPTION
**Pull request by tabrath**
* also upgraded webpack-dev-server and babel-loader and the cli tools are split out to webpack-cli, so installed that.
* updated configuration for new syntax. UglifyJS is now replaced by optimization.minimize flag.
* added 'sideEffects: false' to package.json so webpack can optimize re-exports.

### CODE REVIEW

File: webpack-dev-server.config.js [hunk 1]
```diff
@@ -1,9 +1,10 @@
 const webpack = require("webpack");
 const path = require("path");
 const buildPath = path.resolve(__dirname, "dist");
-const nodeModulesPath = path.resolve(__dirname, "node_modules");
 
 const config = {
+  mode: 'production',
+  target: 'web',
```
Comment (ad1992) [inline webpack-dev-server.config.js]: `target` is `web` by default so can be removed

File: webpack-production.config.js [hunk 2]
```diff
@@ -1,9 +1,10 @@
 const webpack = require("webpack");
 const path = require("path");
 const buildPath = path.resolve(__dirname, "dist");
-const nodeModulesPath = path.resolve(__dirname, "node_modules");
 
 const config = {
+  mode: 'production',
+  target: 'web',
```
Comment (ad1992) [inline webpack-production.config.js]: `target` is `web` by default so can be removed

**Comment by tabrath** [comment]
@ad1992 All done ;)

**Comment by ad1992** [comment]
Thanks for your contribution @tabrath

**Pull request merged**